search/search_repository.go

```diff
@@ -239,6 +245,158 @@ func parseSearchString(rawSearchString string) (searchKeyword, error) {
 	return res, nil
 }
 
+func parseMap(queryMap map[string]interface{}, q *Query) {
+	for key, val := range queryMap {
+		switch concreteVal := val.(type) {
+		case []interface{}:
+			q.Name = key
+			parseArray(val.([]interface{}), &q.Children)
+		case string:
+			q.Name = key
+			s := string(concreteVal)
+			q.Value = &s
+		case bool:
+			s := concreteVal
+			q.Negate = s
+		}
+	}
+}
+
+func parseArray(anArray []interface{}, l *[]Query) {
+	for _, val := range anArray {
+		switch val.(type) {
+		case map[string]interface{}:
+			o := val.(map[string]interface{})
+			q := Query{}
+			parseMap(o, &q)
+			*l = append(*l, q)
+		}
+	}
+}
+
+// Query represents tree structure of the filter query
+type Query struct {
+	// Name can contain a field name to search for (e.g. "space") or one of the
+	// binary operators "$AND", or "$OR". If the Name is not an operator, we
+	// compare the Value against a column in the database that maps to this
+	// Name. We check the Value for equality and for inequality if the Negate
+	// field is set to true.
+	Name  string
+	Value *string
```

Comment:
Operator nodes, with names "$AND", "$OR" etc. will not have values. Since this struct represents tail nodes as well as these operator nodes, the pointer is more suitable.